x-pack/docs/en/security/authentication/security-domain.asciidoc

```diff
@@ -0,0 +1,98 @@
+[role="xpack"]
+[[security-domain]]
+=== Security domains
+
+beta::[]
+
+Security domains are a method of grouping different <<realms,realms>>
+so that the {stack} can recognize when a single user authenticates with
+these realms.
+
+==== How security domains work
+
+Some types of resources in {es} are owned by a single user.
+For example <<async-search,async search contexts>>,
+<<security-api-create-api-key,API keys>>,
+and <<user-profile,user profiles>> each have an owning user.
+
+When a user creates a resource, {es} captures the user's `username` and realm
+information as part of the resource's metadata.
+When the user attempts to access the resource at a later time, {es} compares
+the captured `username` and realm information against those from the accessing
+user. The access is not allowed unless both the realm and `username` match.
+That is, if {es} detects that a `username` from two different realms is
+attempting to access a resource, {es} assumes that these users are distinct and
+doesn't allow resources to be shared between them.
+
+However, there are cases where the same user can authenticate with
+different realms and need to share the same set of resources across the realms.
+For example, an <<ldap-realm,LDAP realm>> and a <<saml-realm,SAML realm>> can
+be backed by the same directory service. Additionally,
+<<configuring-authorization-delegation,authorization delegation>> allows one
+realm to delegate authorization to another realm. If both realms authenticate
+users with the same `username`, it is reasonable to treat these users to
+be the same user.
+
+With security domains, you can configure multiple realms to be under the same
+domain, users can authenticate with any of the realms and have access to the
```

Comment:
```suggestion
With security domains, you group multiple realms to be part of a domain. Users can authenticate with any of the domain realms and have access to the
```

Reply:
I prefer keep saying just `any of the realms` instead of `any of the domain realms` since the meaning is clear enough given the context. Even though it is clear to us what `domain realms` means, it could still be confusing to other readers. So I'd avoid it unless we need to formally introduce the term.
I took other changes. Thanks!